diff --git a/migrate/destinations/MigrateDestinationParagraphsItem.inc b/migrate/destinations/MigrateDestinationParagraphsItem.inc
index 8939040..9fada54 100644
--- a/migrate/destinations/MigrateDestinationParagraphsItem.inc
+++ b/migrate/destinations/MigrateDestinationParagraphsItem.inc
@@ -70,6 +70,9 @@ class MigrateDestinationParagraphsItem extends MigrateDestinationEntity {
 
     $fields['field_name'] = t('Field name');
     $fields['archived'] = t('Archived status of the paragraph item');
+    if (module_exists('uuid')) {
+      $fields['uuid'] = t('UUID');
+    }
 
     $fields += migrate_handler_invoke_all('Entity', 'fields', $this->entityType, $this->bundle, $migration);
     $fields += migrate_handler_invoke_all('ParagraphsItem', 'fields', $this->entityType, $this->bundle, $migration);
diff --git a/paragraphs.install b/paragraphs.install
index a806227..242a10d 100644
--- a/paragraphs.install
+++ b/paragraphs.install
@@ -98,6 +98,16 @@ function paragraphs_schema() {
       ),
     ),
   );
+
+  if (module_exists('uuid')) {
+    $field = uuid_schema_field_definition();
+    $schema['paragraphs_item']['fields']['uuid'] = $field;
+    $schema['paragraphs_item']['indexes']['uuid'] = array('uuid');
+    // This is needed only for revisionable entities.
+    $schema['paragraphs_item_revision']['fields']['vuuid'] = $field;
+    $schema['paragraphs_item_revision']['indexes']['vuuid'] = array('vuuid');
+  }
+
   return $schema;
 }
 
diff --git a/paragraphs.module b/paragraphs.module
index 973b785..3cad328 100644
--- a/paragraphs.module
+++ b/paragraphs.module
@@ -146,6 +146,12 @@ function paragraphs_entity_info() {
     $return['paragraphs_item']['entity cache'] = TRUE;
   }
 
+  if (module_exists('uuid')) {
+    $return['paragraphs_item']['uuid'] = TRUE;
+    $return['paragraphs_item']['entity_keys']['uuid'] = 'uuid';
+    $return['paragraphs_item']['entity_keys']['revision uuid'] = 'vuuid';
+  }
+
   return $return;
 }
 
