diff --git a/url.module b/url.module
index 3f9eba6..242f1d5 100644
--- a/url.module
+++ b/url.module
@@ -364,10 +364,10 @@ function url_fetch_title($url) {
  * Note that #maxlength and #required is validated by _form_validate() already.
  */
 function url_validate_url(&$element, &$form_state) {
-  $value = trim($element['#value']);
-  form_set_value($element, $value, $form_state);
+  // $value = trim($element['#value']);
+  // form_set_value($element, $value, $form_state);

-  if ($value !== '' && !valid_url($value, TRUE)) {
-    form_error($element, t('The URL %url is not valid.', array('%url' => $value)));
-  }
+  // if ($value !== '' && !valid_url($value, TRUE)) {
+  //   form_error($element, t('The URL %url is not valid.', array('%url' => $value)));
+  // }
 }
